external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/OffsetManager.java

```diff
@@ -113,6 +140,15 @@ public long commit(OffsetAndMetadata committedOffset) {
                 break;
             }
         }
+
+        for (Iterator<Long> iterator = emittedOffsets.iterator(); iterator.hasNext();) {
```

Comment:
The emitted offsets have to get deleted at some point, or the set grows indefinitely